File: LeetCodeProblemsGo/LeetCode3516FindClosestPerson/LeetCode3516FindClosestPerson.go
package LeetCode3516FindClosestPerson

import "math"

func findClosest(x int, y int, z int) int {
	delta := math.Abs(float64(x-z)) - math.Abs(float64(y-z))
	if delta > 0 {
		return 2
	}
	if delta < 0 {
		return 1
	}
	return 0
}

File: LeetCodeProblemsCMake/Problems/LeetCode3516FindClosestPerson.cpp
//
// Created by saeed on 9/4/25.
//

#include "LeetCode3516FindClosestPerson.h"

#include <cmath>
#include <math.h>

int LeetCode3516FindClosestPerson::findClosest(int x, int y, int z) {
    int delta=  abs(x - z) - abs(y - z);
    if(delta>0)
        return 2;
    if (delta<0)
        return 1;
    return 0;
}
